test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs

```diff
@@ -0,0 +1,866 @@
+            json.Add("summary", someSummary);
+            json.Add("copyright", someCopyright);
+            json.Add("title", someTitle);
+            json.Add("entryPoint", someEntryPoint);
+            json.Add("projectUrl", someProjectUrl);
+            json.Add("licenseUrl", someLicenseUrl);
+            json.Add("iconUrl", someIconUrl);
+            json.Add("authors", new JArray(authors));
+            json.Add("owners", new JArray(owners));
+            json.Add("tags", new JArray(tags));
+            json.Add("language", someLanguage);
+            json.Add("releaseNotes", someReleaseNotes);
+            var project = GetProject(json);
+
+            project.Description.Should().Be(someDescription);
+            project.Summary.Should().Be(someSummary);
+            project.Copyright.Should().Be(someCopyright);
+            project.Title.Should().Be(someTitle);
+            project.EntryPoint.Should().Be(someEntryPoint);
+            project.ProjectUrl.Should().Be(someProjectUrl);
+            project.LicenseUrl.Should().Be(someLicenseUrl);
+            project.IconUrl.Should().Be(someIconUrl);
+            project.Authors.Should().Contain(authors);
+            project.Owners.Should().Contain(owners);
+            project.Tags.Should().Contain(tags);
+            project.Language.Should().Be(someLanguage);
+            project.ReleaseNotes.Should().Be(someReleaseNotes);
+        }
+
+        [Fact]
+        public void It_sets_the_compilerName_to_csc_when_one_is_not_set_in_the_ProjectJson()
+        {
+            _emptyProject.CompilerName.Should().Be("csc");
+        }
+
+        [Fact]
+        public void It_sets_the_compilerName_to_the_one_in_the_ProjectJson()
+        {
+            const string compilerName = "a compiler different from csc";
+            var json = new JObject();
+            json.Add("compilerName", compilerName);
+            var project = GetProject(json);
+
+            project.CompilerName.Should().Be(compilerName);
+        }
+
+        [Fact]
+        public void It_leaves_testRunner_null_when_one_is_not_set_in_the_ProjectJson()
+        {
+            _emptyProject.TestRunner.Should().BeNull();
+        }
+
+        [Fact]
+        public void It_sets_testRunner_to_the_one_in_the_ProjectJson()
+        {
+            const string someTestRunner = "some test runner";
+            var json = new JObject();
+            json.Add("testRunner", someTestRunner);
+            var project = GetProject(json);
+
+            project.TestRunner.Should().Be(someTestRunner);
+        }
+
+        [Fact]
+        public void It_sets_requireLicenseAcceptance_to_false_when_one_is_not_set_in_the_ProjectJson()
+        {
+            _emptyProject.RequireLicenseAcceptance.Should().BeFalse();
+        }
+
+        [Fact]
+        public void It_sets_requireLicenseAcceptance_to_the_one_in_the_ProjectJson()
+        {
+            var json = new JObject();
+            json.Add("requireLicenseAcceptance", true);
```

Comment:
test name is a bit misleading. All this proves is that the value is true when a value is set in the Json...